app/models/shipit/stack.rb

```diff
@@ -398,6 +398,15 @@ def to_param
       [repo_owner, repo_name, environment].join('/')
     end
 
+    def self.run_deploy_in_foreground(stack:, revision:)
+      stack = Shipit::Stack.from_param!(stack)
+      until_commit = stack.commits.where(sha: revision).limit(1)
+      env = stack.deploy_variables.map(&:name)
+      current_user = Shipit::AnonymousUser.new
```

Comment:
I wonder if we should create a `CLIUser` or similar, so it's explicit where the deploy was triggered.